packages/graphql-cli-validate-queries/__tests__/index.spec.js

```diff
@@ -0,0 +1,222 @@
+        eslintCLIEngineSpy.mockClear();
+        existsSyncSpy.mockClear();
+        mockConsoleLog.mockClear();
+        mockConsoleWarn.mockClear();
+        mockProcessExit.mockClear();
+    });
+    afterAll(() => {
+        eslintCLIEngineSpy.mockRestore();
+        existsSyncSpy.mockRestore();
+        mockConsoleLog.mockRestore();
+        mockConsoleWarn.mockRestore();
+        mockProcessExit.mockRestore();
+    });
+
+    test('it is a function', () => {
+        expect(plugin.handler).toBeInstanceOf(Function);
+    });
+
+    test('it returns undefined', async () => {
+        const actual = await plugin.handler(mockContext, mockArgs);
+
+        expect(actual).toBeUndefined();
+    });
+
+    test("it throws if the schema doesn't exist locally", async () => {
+        // Mock the file not existing.
+        existsSyncSpy.mockImplementationOnce(() => false);
+
+        await plugin.handler(mockContext, mockArgs);
+
+        expect(existsSyncSpy).toHaveBeenCalled();
+        expect(mockContext.spinner.fail).toHaveBeenCalled();
+        expect(mockProcessExit).toHaveBeenCalledWith(1);
+    });
+
+    test('it creates a validator with the correct configuration', async () => {
+        const expectedRule = [
+            'error',
+            // These objects are derived from mockArgs.
+            {
+                env: 'apollo',
+                projectName: 'myApp'
+            },
+            {
+                env: 'literal',
+                projectName: 'myApp'
+            }
+        ];
+
+        await plugin.handler(mockContext, mockArgs);
+
+        const lintConfiguration = eslintCLIEngineSpy.mock.calls[0][0];
+
+        const keys = Object.keys(lintConfiguration);
+        expect(keys).toHaveLength(4);
+        expect(keys).toContain('parser');
+        expect(keys).toContain('plugins');
+        expect(keys).toContain('rules');
+        expect(keys).toContain('useEslintrc');
+
+        expect(lintConfiguration.parser).toBe('babel-eslint');
+
+        expect(lintConfiguration.plugins).toHaveLength(1);
+        expect(lintConfiguration.plugins).toContain('graphql');
+
+        const rulesKeys = Object.keys(lintConfiguration.rules);
+        expect(rulesKeys).toHaveLength(2);
+        expect(rulesKeys).toContain('graphql/template-strings');
+        expect(rulesKeys).toContain('graphql/no-deprecated-fields');
+
+        const templateStringsRule =
+            lintConfiguration.rules['graphql/template-strings'];
+        expect(templateStringsRule).toEqual(expectedRule);
+
+        const deprecatedFieldsRule =
+            lintConfiguration.rules['graphql/no-deprecated-fields'];
+        expect(deprecatedFieldsRule).toEqual(expectedRule);
+
+        expect(lintConfiguration.useEslintrc).toBe(false);
+    });
+
+    test('it logs an appropriate message when there are no errors', async () => {
+        await plugin.handler(mockContext, mockArgs);
+
+        expect(mockConsoleLog).toHaveBeenCalledWith('All queries are valid.');
```

Comment:
Using the `if (module === require.main) {}` trick, you can omit a lot of this mocking.